### PR TITLE
Added support for floating point knob values

### PR DESCRIPTION
**Pull request by brad-kelly**
This PR adds support for floating point knob values

### CODE REVIEW

File: framework/logging.cpp [hunk 1]
```diff
@@ -1453,6 +1453,10 @@ void logging_mark_knob_used(std::string_view key, TestKnobValue value, KnobOrigi
             else
                 return stdprintf("-0x%" PRIx64, -v);
         }
+        std::string operator()(double v)
+        {
+            return stdprintf("%f", double(v));
```
Comment (thiagomacieira) [inline framework/logging.cpp:1458]: I recommend "%.17g" here. Also possible, "%.*g" with a first argument of `std::numeric_limits<double>::digits10 + 2`.

Why 17 or that +2? Long story...

File: framework/test_knobs.h [hunk 2]
```diff
@@ -31,6 +31,8 @@ uint64_t get_testspecific_knob_value_uint(const struct test *test, const char *k
                                           uint64_t value_if_not_present);
 int64_t  get_testspecific_knob_value_int(const struct test *test, const char *key,
                                          int64_t value_if_not_present);
+double   get_testspecific_knob_value_double(const struct test *test, const char *key,
```
Comment (thiagomacieira) [inline framework/test_knobs.h:34]: Can you add the macro in the `RESTRICTED_CMDLINE` section above? (GitHub doesn't let me comment there).

**Review by thiagomacieira — CHANGES_REQUESTED** [review]
Looks mostly good.

Can you also add a selftest to bats/sanity-check/20-selftests.bats? Modify `selftest_logs_options`

File: framework/logging.cpp (hunk 1 above)
Comment (brad-kelly) [reply framework/logging.cpp]: Used the %.17g"  Just seems easier to read

File: framework/test_knobs.h (hunk 2 above)
Comment (brad-kelly) [reply framework/test_knobs.h:34]: Done.

**Review by thiagomacieira — APPROVED** [review]

**Pull request merged**